engine/grid.py

```diff
@@ -691,21 +739,35 @@ def get_all_guided_lawnmower_waypoints_adjustable(self, is_vertical):
         """
         # TODO: move find border nodes into this func instead of doing it
         # everytime before this is called
-        self.curr_pos = self.leftmost_node_pos
-        self.waypoints.append(self.curr_pos)
-        # TODO: Change to allow choosing down and left directions
+        # self.curr_pos = self.bottom_left_pos
         # TODO: Implement function to traverse vertical triangle/ slanted shapes
-        if is_vertical:
-            self.direction = self.direction.UP
+        if direction == "DOWN":
+            self.direction = self.Direction.DOWN
+        elif direction == "UP":
+            self.direction = self.Direction.UP
+        elif direction == "LEFT":
+            self.direction = self.Direction.LEFT
+        else:
+            self.direction = self.Direction.RIGHT
```

Comment:
might be better to have `elif direction == "RIGHT"` and then have an else condition that throws an error